apps/web/app/(app)/gardens/[chain]/[garden]/pool/[poolId]/proposals/[proposalId]/page.tsx

```diff
@@ -53,168 +54,210 @@ const prettyTimestamp = (timestamp: number) => {
   return `${day} ${month} ${year}`;
 };
 
-export default async function Proposal({
+export default function Proposal({
   params: { proposalId, poolId, chain, garden },
 }: {
   params: { proposalId: string; poolId: number; chain: number; garden: string };
 }) {
   // TODO: fetch garden decimals in query
-  const { data: getProposalQuery } = await queryByChain<getProposalDataQuery>(
-    urqlClient,
-    chain,
-    getProposalDataDocument,
-    {
-      garden: garden,
-      proposalId: proposalId,
-    },
-  );
+  const { data: getProposalQuery } =
+    useSubgraphQueryByChain<getProposalDataQuery>(
+      chain,
+      getProposalDataDocument,
+      {
+        garden: garden,
+        proposalId: proposalId,
+      },
+      {},
+      { topic: "proposal", id: proposalId, type: "update", chainId: chain },
```

Comment:
should we use constants to TOPIC and TYPE?

Reply:
Both are strongly type with template literal so I would preffer saving us an extra import 😅

Reply:
saving for what? its not low level coding. but yeah if its strongly type we are safe.

Reply:
savving an extra step